design/specs/column-extraction.md

```diff
@@ -8,87 +8,58 @@ editor: markdown
 
-## Selecting Columns
-Columns must be given a new interaction in order to be selected. A column header can be selected by clicking on it, as in spreadsheet applications. Multi-selection is also possible by holding down the shift key and clicking on multiple column headers. Clearing the selection is as simple as clicking anywhere outside of the selection target or on a selected column once more.
+This spec describes the steps a user would take to create a new table from a subset of columns from an existing table.
 
-## Selection Actions
-Depending on a particular selection, the inspector panel will display actions that can be performed on the selection, as described above with columns. These selection actions will be determined by the type and number of items chosen.
+## Design Goals
 
-# User Flow
+- Allow users to split a table into multiple new tables by selecting columns
+- Help users understand the impact of splitting a table
+- Help users understand how the tables are linked after the split operation
 
-1. **The user imports a table from a file**
+## Splitting a Table by Extracting Columns
 
-    This user flow assumes that the starting point is an imported table containing columns that can be extracted and turned into entities. A column that may be transformed into a table or entity will usually include other columns that are attributes for that entity. If the database is not normalized, the entity to be extracted might include duplicated values. For example, consider a table of publications with multiple entries for the same publisher in the publisher column.
+### Selecting columns as a starting point
 
-2. **The user selects the columns (attributes) for the new entity**
+The user will select one or multiple columns to start with the column extraction process. All columns can be extracted, except for primary key columns. Once selected, the inspector panel will list 'Extract Columns to Table' as an action. Selecting this option will open the 'Extract Columns to Table' dialog.
 
-    The user chooses all the columns that will be attributes for the new object. For example, in the instance of the publisher, there could be columns for 'Publisher Name' and 'Publisher Id.' Including all the attributes is very important since columns cannot be moved after the fact.
+![image](/assets/design/specs/column-extraction/185139636-9a8a048c-9f87-4e26-aa2c-88f4e8d1648d.png)
 
-     <img width="1512" alt="image" src="https://user-images.githubusercontent.com/845767/177544077-9f074830-ce64-4ed9-80bc-53de53977f9c.png">
+## Extract Columns Dialog
 
+The dialog will list the columns that will be extracted and the table that will be created. The user can change the name of the new table. The dialog will also list the foreign key that will be created between the new table and the original table. The user can change the name of the foreign key.
```

Comment:
This presumes that we will always move columns to a new table. We also support moving columns to an existing table (and we need this for the demo, please see Screen 15 in the [Storyboard](https://www.figma.com/file/F0FmNaNz8hvrgxPax3Bix1/Cycle-3-reference?node-id=0%3A1))

Reply:
I think in this case 'moving columns' would apply, and the available tables would have to be linked already.

Reply:
Yes. The spec needs to cover this case, though, since it's required for the storyboard. Please see my note above about two separate actions.